File: matlab_bridge/matlab_unittest/t_general_species.m
function t_general_species(verbosity)

% TESTS fields stored in general/subject

script_base_name = mfilename();
script_name = [script_base_name '.m'];
fname = [regexprep(script_base_name, '^t_', 's_') '.nwb'];
if nargin < 1
    % default, display all information. Other options are: 'none', 'summary'
    verbosity = 'all';
end
        
function test_field(fname, name)
    val = test_utils.verify_present(fname, 'general/subject/', lower(name));
    if ~strcmp(val, name)
        test_utils.error('Checking metadata', 'field value incorrect');
    end
end

function test_general_subject()
    create_general_subject(fname);
    val = test_utils.verify_present(fname, 'general/subject/', 'description');
    if ~strcmp(val, 'SUBJECT')
        ut.error('Checking metadata', 'field value incorrect')
    end
    test_field(fname, 'SUBJECT_ID');
    test_field(fname, 'SPECIES');
    test_field(fname, 'GENOTYPE');
    test_field(fname, 'SEX');
    test_field(fname, 'AGE');
    test_field(fname, 'WEIGHT');
end


function create_general_subject(fname)
    settings = {'file_name', fname, 'mode', 'w', 'verbosity', verbosity, ...
        'identifier', nwb_utils.create_identifier('general subject test'), ...
        'description','test elements in /general/subject'};
    f = nwb_file(settings{:});
    
    %
%     neurodata.set_metadata(SUBJECT, 'SUBJECT')
%     neurodata.set_metadata(SUBJECT_ID, 'SUBJECT_ID')
%     neurodata.set_metadata(SPECIES, 'SPECIES')
%     neurodata.set_metadata(GENOTYPE, 'GENOTYPE')
%     neurodata.set_metadata(SEX, 'SEX')
%     neurodata.set_metadata(AGE, 'AGE')
%     neurodata.set_metadata(WEIGHT, 'WEIGHT')
    %
    
    g = f.make_group('subject');
    g.set_dataset('description', 'SUBJECT');
    g.set_dataset('subject_id', 'SUBJECT_ID');
    g.set_dataset('species', 'SPECIES');
    g.set_dataset('genotype', 'GENOTYPE');
    g.set_dataset('sex', 'SEX');
    g.set_dataset('age', 'AGE');
    g.set_dataset('weight', 'WEIGHT');
    
    % neurodata.close()
    f.close()
end

test_general_subject()
fprintf('%s PASSED\n', script_name);
end

File: unittest/t_general_species.py
#!/usr/bin/python
# import nwb
# from nwb.nwbco import *
import test_utils as ut

from nwb import nwb_file
from nwb import nwb_utils as utils

# TESTS fields stored in general/subject

def test_field(fname, name):
    val = ut.verify_present(fname, "general/subject/", name.lower())
    if not ut.strcmp(val, name):
        ut.error("Checking metadata", "field value incorrect")

def test_general_subject():
    if __file__.startswith("./"):
        fname = "s" + __file__[3:-3] + ".nwb"
    else:
        fname = "s" + __file__[1:-3] + ".nwb"
    create_general_subject(fname)
    val = ut.verify_present(fname, "general/subject/", "description")
    if not ut.strcmp(val, "SUBJECT"):
        ut.error("Checking metadata", "field value incorrect")
    test_field(fname, "SUBJECT_ID")
    test_field(fname, "SPECIES")
    test_field(fname, "GENOTYPE")
    test_field(fname, "SEX")
    test_field(fname, "AGE")
    test_field(fname, "WEIGHT")


def create_general_subject(fname):
    settings = {}
    settings["file_name"] = fname
    settings["identifier"] = utils.create_identifier("general top test")
    settings["mode"] = "w"
    settings["description"] = "test top-level elements in /general"
    settings["verbosity"] = "none"
    f = nwb_file.open(**settings)
    
    
    #
#     neurodata.set_metadata(SUBJECT, "SUBJECT")
#     neurodata.set_metadata(SUBJECT_ID, "SUBJECT_ID")
#     neurodata.set_metadata(SPECIES, "SPECIES")
#     neurodata.set_metadata(GENOTYPE, "GENOTYPE")
#     neurodata.set_metadata(SEX, "SEX")
#     neurodata.set_metadata(AGE, "AGE")
#     neurodata.set_metadata(WEIGHT, "WEIGHT")
    #
    
    g = f.make_group("subject")
    g.set_dataset("description", "SUBJECT")
    g.set_dataset("subject_id", "SUBJECT_ID")
    g.set_dataset("species", "SPECIES")
    g.set_dataset("genotype", "GENOTYPE")
    g.set_dataset("sex", "SEX")
    g.set_dataset("age", "AGE")
    g.set_dataset("weight", "WEIGHT")
    
    # neurodata.close()
    f.close()

test_general_subject()
print("%s PASSED" % __file__)

